students/stan_slov7/lesson04/mailroom2.py

```diff
@@ -0,0 +1,212 @@
+                   "line_break" : '\n\n',
+                   "greeting" : "Have a positively wonderful day!",
+                   "signature_indent" : ' '*25,
+                   "sender_indent" : ' '*20,
+                   "sender_name" : "Your Friendly Neighborhood CEOs"
+                  }
+
+
+    added_donation_template = "\n".join(("\n\tNew Donation Amount of: ${donation_amount:.2f} has been recorded in the donor database,",
+                              "under the associated Donor Name: {donor_name}. Thanks for contributing!{line_break}")).format(**layout_dict)
+    
+    email_individual_template = "\n".join(("Dear {donor_name},{line_break}",
+                                "{new_pgph_indent}Thank you for the generous donation of: ${donation_amount:.2f}!{line_break}",
+                                "You will forever hold a place in our hearts! {greeting}{line_break}",
+                                "{sender_indent}Sincerely,{line_break}",
+                                "{signature_indent} - {sender_name}{line_break}")).format(**layout_dict) 
+    
+    email_all_template = "\n".join(("Dear {donor_name},{line_break}",
+                         "{new_pgph_indent}Thank you for your repeated historical donations totalling: ${donation_amount:.2f}!{line_break}",
+                         "Your long time generosity recognized you as a pillar of our community!{line_break}",
+                         "{greeting}{line_break}",
+                         "{sender_indent}Sincerely,{line_break}",
+                         "{signature_indent} - {sender_name}{line_break}")).format(**layout_dict) 
+    
+    
+    templates = {1 : added_donation_template,
+                 2 : email_individual_template,
+                 3 : email_all_template
+                }
+    
+    return templates[template_num]
+
+
+def sort_key(hist_tup):
+    return hist_tup[1]   #sorted by total (historical) donation amount
+
+
+def quit_program():
+    print("Thank you, this program will now terminate.")
+    sys.exit()
+
+
+#Updated for dicts: redirect to functions based on selection using new dict structure, and call function from dict
+def main():
+    
+    os.chdir(current_dir)
+    #To ensure files get written in current working directory
+    
+    while True:
+        response = int(input(main_prompt))  
+        if response in main_menu_dict:
+            main_menu_dict.get(response)()
+            #fetches relevant value based on key selection which is a call to the associated feature function
+        else:
+            print(response, " is not a valid selection. Please enter a valid menu option value (1 - 4).\n")
+
+
+main_menu_dict = {1 : send_thank_you,
+                  2 : create_report,
+                  3 : write_email_all,
+                  4 : quit_program
+                  }
+
+
+if __name__ == "__main__":
+    
+    # block to guard against your code running automatically if this module is imported
+    main()
```

Comment:
code looks great! very well done and nice use of functions to modularize your code.